core/modules/wdb/ChunkResource.cc

```diff
@@ -235,8 +279,112 @@ class Backend {
         }
     }
 
+    /// Run the 'query'. If it fails, terminate the program.
+    void _execLockSql(std::string const& query) {
+        LOGF_DEBUG("execLockSql %1%" % query);
+        sql::SqlErrorObject err;
+        if(!_sqlConn.runQuery(query, err)) {
+            LOGF_ERROR("Lock failed, exiting. query=%1%  err=%2%" % query % err.printErrMsg());
+            exit(EXIT_FAILURE);
+        }
+    }
+
+    /// Return the status of the lock on the in memory tables.
+    LockStatus _memLockStatus() {
+        std::string sql = "SELECT uid FROM " + _lockDbTbl + " WHERE keyId = 1;";
+        sql::SqlResults results;
+        sql::SqlErrorObject err;
+        if (!_sqlConn.runQuery(sql, results, err)) {
+            // Assuming UNLOCKED should be safe as either it must be LOCKED_OURS to continue
+            // or we are about to try to lock. Failure to lock will cause the program to exit.
+            std::ostringstream emsg;
+            emsg << "memLockStatus query failed, assuming UNLOCKED. " << sql << " err=" << err.printErrMsg();
+            LOGF_WARN("%1%" % emsg.str());
```

Comment:
Ah I see - that makes sense. This makes me wish that log provided macros that worked with streams instead of boost::format. (I know, I know - too many macros already).